### ISSUE TITLE
sap_swpm: nw_java_import.buildJEEusingExtraMileTool templated as "True"

### ISSUE THREAD

**Issue by Galaxy8977** [sap-linuxlab/community.sap_install#915]
When installing SAP NW750 JAVA or SOLMAN72SR2 JAVA instances,

In roles/sap_swpm/templates/inifile_params.j2, the nw_java_import.buildJEEusingExtraMileTool is templated from the follow line:

```
nw_java_import.buildJEEusingExtraMileTool = {{ true if sap_swpm_java_import_method == 'extramile' else false }}
```

This leads to the result of 

```
nw_java_import.buildJEEusingExtraMileTool = True
```

And SWPM actually think it is not `true` and then going to the `Deploy Controller Runner tool` logic, the sap_swpm role eventually errored out with the described error in `docs/developer_notes/sap_swmp_dev.md`.

```log
#### Error with the Deploy Controller Runner tool

When setting inifile.params `nw_java_import.buildJEEusingExtraMileTool = false` an error will occur with the Deploy Controller Runner tool:

[ LOG EXCEPTION ]  :: Exception while generating ID: [[ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized]
com.sap.engine.services.dc.cm.DCNotAvailableException: [ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized

[ Error ] Exception while generating ID: [[ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized]

[ LOG EXCEPTION ]  :: com.sap.engine.services.dc.api.deploy.DeployException: [ERROR CODE DPL.DCAPI.1023] DCNotAvailableException.
Reason: [ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized
Caused by: com.sap.engine.services.dc.cm.DCNotAvailableException: [ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized

[ LOG EXCEPTION ]  :: java.lang.NullPointerException: while trying to get the length of a null array loaded from a local variable at slot 4

[ Error ]
104
Component:n/a
Status:Initial
Description:
DeployException: [ERROR CODE DPL.DCAPI.1023] DCNotAvailableException.
Reason: [ERROR CODE DPL.DC.3218] Deploy controller is not ready to serve yet. Probably a un/deploy operation after offline phase is in progress at the moment. DC state is Initialized
```

Overriding `nw_java_import.buildJEEusingExtraMileTool` to `true` using `sap_swpm_inifile_parameters_dict` explicitly can workaround this, so I believe it is the value `True` can not be regarded as `true` by SWPM.

**Comment by berndfinger** [comment]
Solved in #916.

**Comment by berndfinger** [comment]
@Galaxy8977 Many thanks for reporting this issue! It should be fixed now.
